src/Ocelot.Provider.Consul/OcelotBuilderExtensions.cs

```diff
@@ -24,9 +24,8 @@ public static IOcelotBuilder AddConsul(this IOcelotBuilder builder)
     {
         builder.Services
             .AddSingleton(ConsulProviderFactory.Get)
-            .AddSingleton(ConsulProviderFactory.GetConfiguration)
             .AddSingleton<IConsulClientFactory, ConsulClientFactory>()
-            .AddSingleton<IConsulServiceBuilder, DefaultConsulServiceBuilder>()
+            .AddScoped<IConsulServiceBuilder, DefaultConsulServiceBuilder>()
             .RemoveAll(typeof(IFileConfigurationPollerOptions))
             .AddSingleton<IFileConfigurationPollerOptions, ConsulFileConfigurationPollerOption>();
```

Comment:
@ggnaegi Fascinating design, isn't it? 😄 We're upgrading from the In-Memory poller to this advanced one. Exciting!